### DISCUSSION TITLE
What do people use for pre- and post-processing?

### DISCUSSION THREAD

**Discussion by yw5aj** [OpenRadioss/OpenRadioss#404] [Users]
Hi all,

Very excited to hear the news of Radioss now is open source!

Just wondering, what do people usually use for pre- and post-processing?

Shawn

**Comment by laurent-altr** [comment]
Welcome Shawn, 

[Here](https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/21397510/Pre+and+Post+Processing+for+OpenRadioss) is an article discussing this topic.

Maybe @MarianBulla can add more comments on this topic. 

Laurent

**Reply by JiancaiZheng** [reply]
Hi Laurent,
The document is not clear about the post-processing. Could you provide some suggestions about how to convert the animation and timehistory files to VTK format result files?

Jiancai Zheng

**Reply by MarianBulla** [reply]
Hi Jiancai,
thank you very much for your remark. It's a very good point. 
We are currently finalizing a converter for the animation (convert to VTK) and the timehistory file (convert to csv). 
Doing final tests, to be sure it works.
We will provide it to GitHub within the next few days, which will make you able to convert the OpenRadioss results for ParaView.

Hope it helps you ?

Thanks and best regards,
Marian

**Reply by JiancaiZheng** [reply]
Hi Laurent,
Thank you for your replay. Meanwhile, we are looking forward to the official provision of converters as soon as possible.

Jiancai Zheng

**Comment by MarianBulla** [comment]
Hi Shawn,
thank you very much for your nice comment.
Regarding pre- please have a look at the recent gmsh version. 
OpenRadioss format output is already there.

Thanks and best regards,
Marian

**Reply by yw5aj** [reply]
Thank you for your help Marian!

**Reply by paul18frr** [reply]
`Regarding pre- please have a look at the recent gmsh version.`
Hi
We're speaking about the development release, not the stable one  ;-)
Paul

**Comment by dragazsolt** [comment]
Meshing: Hypermesh or Ansa
Model built: Hypercrash

**Comment by dragazsolt** [comment]
LS- Dyna Pre- post could be alternative option. It is also free to setup Dyna model.

**Comment by huisky2000** [comment]
What is the status now? I'm still not sure how to view the results. The animation and time history are not enough, how to view the results with stress and strain outputs?

**Reply by MarianBulla** [reply]
Hi @huisky2000 , have a look in the solver documentaion and Reference Guide here:
https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/4816906/OpenRadioss+User+Documentation
and here for online:
https://2022.help.altair.com/2022.1/hwsolvers/rad/topics/solvers/rad/overview_ref_guide_rad_c.htm
You can add the requests for additional output into the animation files in the engine section and addtional plots in the /TH/ in the starter section.
Hope this helps you ?

**Comment by dekanewton** [comment]
Hi,
Is it possible to create named element and node sets with GMSH to apply boundary conditions and material properties based on sets? Is there any tutorial how to do this? Available tensile test example does not shown this.
Thanks.

OK,
It looks that GMSH is able to create sets for use in OpenRadioss... I just need to find some time to start play with it :)

**Reply by PaulAltair** [reply]
I'm not sure how this can be achieved (I wrote both the Radioss export template for GMSH and the tensile test tutorial using GMSH, and I didn't code anything for generating node/element sets!) I'm not very experienced with GMSH, if there was a GMSH entity I could map, even to the most basic /GRNOD/NODE type definition I could try and implement it but so far I didn't find a way to create a set on the GMSH side

**Reply by PaulAltair** [reply]
ok, I've had a dig in gmsh source, currently, if one defines 'physical groups' on e.g. a surface prior to meshing, then mesh, and on export you select 'save groups of elements' or 'save groups of nodes' then you will get at the end of your deck, node and element groups in LS-DYNA format, comma separated. This functionality, at least gives you the entities you need, that you can then manipulate in text editor, but I will revisit the gmsh source to see if I can update this to instead write Radioss format groups to make life easier!

**Reply by dragazsolt** [reply]
LS - PREPOST is a free from Dynamore and Ansys.   I think it is a really amazing software. 
High quality mesh can be generated by LS-Prepost.

**Reply by PaulAltair** [reply]
> ok, I've had a dig in gmsh source, currently, if one defines 'physical groups' on e.g. a surface prior to meshing, then mesh, and on export you select 'save groups of elements' or 'save groups of nodes' then you will get at the end of your deck, node and element groups in LS-DYNA format, comma separated. This functionality, at least gives you the entities you need, that you can then manipulate in text editor, but I will revisit the gmsh source to see if I can update this to instead write Radioss format groups to make life easier!

I have updated the export template for GMSH and created a pull request, my modified template will write node and element groups from GMSH PhysicalGroup definitions on export, if the correct options are chosen (save groups of elements, save groups of nodes), if using this approach then the mesh export options also need to be toggled to 'Save all' otherwise ONLY those groups will be written. Also, because Radioss format needs separate groups for 4 Node Shells and Trias, the export template currently writes both, one of which will be redundant (or bits of both will be if both element types are in the groups)

Update will initially only be in the gmsh development build, now merged (since 11th Jun)

**Reply by PaulAltair** [reply]
I created a tutorial demoing simple usecase for this [Tube Contact Tutorial using Gmsh](https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/54624257/Tubes+Contact+Tutorial+Using+Gmsh)

**Reply by dekanewton** [reply]
Wow, that's great! Thank you.

**Reply by PaulAltair** [reply]
I hadn't posted here, but in latest gmsh dev build since last week some time, I also updated it further so now it only writes the SHEL or SH3N with correct content as required for the surface groups

**Comment by bobim** [comment]
Hi there,

I just found out about OpenRadioss 2-3 months ago and I'm still ramping up on it, but wanted to share my workflow:

Geometry and mesh with [Salome](https://www.salome-platform.org/)
Exporting meshes with a simple python script
Reusing exported mesh from another python script to generate the rad files

I initially wanted to use [operato](https://github.com/nawijn/operato) but couldn't grok how to implement the missing keywords, so went the simple route of writing cards from the python script. Here is a working example: [forming-example.tar.gz](https://github.com/OpenRadioss/OpenRadioss/files/12061204/forming-example.tar.gz), with a Salome database, the export and input deck-building scripts. The export script iterates over all meshes and outputs a single json file with a dictionary containing a nodes list, elements by part, and groups for boundary conditions, loads, interfaces. Maybe that would be clever to export each mesh separately in order to be able to play with their positioning, e.g. to adjust tool closing for different blank thicknesses. Another route could be exporting individual meshes to med files from Salome and leverage the [medcoupling](https://www.salome-platform.org/?page_id=394)  library to read back mesh information in the input deck-building python script, with niceties like using fields to model shell thickness or temperature distributions.

Cheers,
Etienne
